code/datums/components/storage/storage.dm

```diff
@@ -524,8 +524,8 @@ GLOBAL_LIST_EMPTY(cached_storage_typecaches)
 //Tries to dump content
 /datum/component/storage/proc/dump_content_at(atom/dest_object, mob/M)
 	var/atom/A = parent
-	var/atom/dump_destination = dest_object.get_dumping_location()
-	if(A.Adjacent(M) && dump_destination && M.Adjacent(dump_destination))
+	var/atom/dump_destination = get_dumping_location(dest_object)
+	if(M.CanReach(A) && dump_destination && M.CanReach(dump_destination))
```

Comment:
Doesn't this enable dumping outside of locked lockers etc ?

Reply:
na